packages/dds/map/src/directory.ts

```diff
@@ -585,10 +587,24 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
     }
 
     /**
-     * {@inheritDoc @fluidframework/shared-object-base#SharedObject.snapshot}
+     * {@inheritDoc @fluidframework/shared-object-base#SharedObject.snapshotCore}
      */
-    public snapshot(): ITree {
-        return serializeDirectory(this.root);
+    public snapshot(): IChannelSnapshotDetails {
```

Comment:
Consider following change:
Today, DDS is the only layer that uses snapshot format, all other layers use summaries. We need to change that.
I'd create summarize() method with new signature, and implement one in SharedObject that would
- create serizlizer
- call into snapshot()
- convert snapshot into summary (convertToSummaryTree())
- return IChannelSnapshotDetails  (rename it to IChannelSummaryDetails).

And make sure that whatever returned from DDS is in exactly same format (same interface) as the one data store runtime would use to return to container runtime - there should be no differences!

Reply:
That way snapshot() impls can stay always the same, just need to use serializer passed to them.

Reply:
Added serializer API to DDSs and made the return type same across all context<->channel interfaces.